pgjdbc/src/main/java/org/postgresql/core/AuthenticationPluginManager.java

```diff
@@ -67,14 +71,14 @@ private AuthenticationPluginManager() {
    * @throws PSQLException Throws a PSQLException if the plugin class cannot be instantiated or if the retrieved password is null.
    */
   public static byte[] getEncodedPassword(AuthenticationRequestType type, Properties info) throws PSQLException {
-    String password = getPassword(type, info);
+    char[] password = getPassword(type, info);
 
     if (password == null) {
       throw new PSQLException(
           GT.tr("The server requested password-based authentication, but no password was provided."),
           PSQLState.CONNECTION_REJECTED);
     }
 
-    return password.getBytes(StandardCharsets.UTF_8);
+    return String.valueOf(password).getBytes(StandardCharsets.UTF_8);
```

Comment:
We need to clarify the encoding in the javadoc as it is a part of public API.

Please erase the contents of `password` array after the conversion.
I guess `Charset.encode()` should be good for converting the data without creating an immutable `String`:

```java
ByteBuffer buf = StandardCharsets.UTF_8.encode(CharBuffer.wrap(toBeEncoded));
byte[] array = new byte[buf.limit()];
buf.get(array);
```

See https://stackoverflow.com/a/33791944/1261287

Reply:
I don't think we should touch the `char[] password` array after usage. It's not ours to destroy as the user provided it to us and it's likely they will have a single value of the array available for reuse in multiple connections say from a connection pool. If it should be cleared then the caller as the owner of that data is responsible for it.

Reply:
We probably need to document the ownership of the password arrays.
I think we do not need the password after we connect.

Reply:
For the `String.valueOf(...)` replacement, do you mean like this:

```java
    ByteBuffer buf = StandardCharsets.UTF_8.encode(CharBuffer.wrap(password));
    byte[] encodedBytes = new byte[buf.limit()];
    buf.get(encodedBytes);
    return encodedBytes;
```

Reply:
Yes I don't think we need the password after we've used it during the connect handshake. The only spot that might use it is query cancellation but we do it via faster pid + secret so the password is not required. And even the "getURL()" type methods wouldn't need it as the user has already indicated the password is being provided on the fly.

Reply:
Erasing the password completely is a different feature, and we probably need to roll it under a feature flag first.

However, I am inclined we should treat `char[] password` as one-time use only containers: if we pass them somewhere, we expect they will be released after use.
That is why I think erasing `char[]` after we get `encodedBytes` makes sense: we are going to use `encodedBytes` anyway, so why keep `char[]`?